components/sup/src/manager/service/mod.rs

```diff
@@ -732,11 +706,6 @@ impl Service {
         Ok(())
     }
 
-    fn run_hook(&mut self, hook: HookType) -> Result<()> {
-        self.hooks.compile(&self.service_group, &self.config);
-        self.hooks.try_run(hook, &self.service_group)
-    }
-
     /// Update our own service rumor with a new configuration from the packages exported
```

Comment:
Turns out I have learned that changes to census data results in a config change and triggers a reconfigure which clears it all up for me and makes me fine with removing recompile before every hook execution.